packages/stark-core/src/common/bootstrap/abstract-stark-main.ts

```diff
@@ -0,0 +1,157 @@
+import {ApplicationRef, ComponentRef, enableProdMode, NgModuleRef} from "@angular/core";
+import {disableDebugTools, enableDebugTools} from "@angular/platform-browser";
+import {createNewHosts} from "@angularclass/hmr";
+
+import {StarkMain} from "./stark-main.intf";
+import {StarkEnvironment} from "environments/environment";
+
+/**
+ * Parent class for bootstrapping Stark applications.
+ * It handles the DOMContentLoaded event of the Web browser in order to bootstrap the application.
+ * This class also supports HMR in development mode.
+ */
+export abstract class AbstractStarkMain implements StarkMain {
+	protected constructor(protected environment: StarkEnvironment) {
+		// no-op
+	}
+
+	/**
+	 * Must be implemented by children classes. Called within bootstrapDomReady (i.e., when the DOM has been loaded)
+	 */
+	public abstract main: () => Promise<any>;
+
+	public mainWrapper = (): void => { // written like this because otherwise "this" will not be captured :)
+		// no need for the event listener after this
+		document.removeEventListener("DOMContentLoaded", this.mainWrapper, false);
+
+		if (this.environment.hmr) {
```

Comment:
I'm wondering if it is better to change this check to use the ambient ENV variable. This way the code inside will be removed in PROD. But in order for this to works, then we should also add this check inside the `bootstrapHmr` method so that the actual usage of the `angularclass/hmr` module is also removed from the bundle.

What do you think?

Reply:
Indeed, let's keep it light :)